src/EFCore.PG/Query/Expressions/Internal/CustomBinaryExpression.cs

```diff
@@ -60,7 +61,9 @@ public class CustomBinaryExpression : Expression
         protected override Expression Accept(ExpressionVisitor visitor)
             => visitor is NpgsqlQuerySqlGenerator npgsqlGenerator
                 ? npgsqlGenerator.VisitCustomBinary(this)
-                : base.Accept(visitor);
+                : visitor is NpgsqlSqlTranslatingExpressionVisitor npgsqlVisitor
```

Comment:
This seems very odd and non-standard, and is probably related to this PR deviating from the standard translation -> generation pipeline. `CustomBinaryExpression` is something that's only supposed to exist in the SQL generation phase (which is rather late), and not in the translation phase.

Reply:
I think you're right, I'll give this another look too.

Reply:
See my comments above for `ArrayAnyAllExpression`.